check_limits.py

```diff
@@ -1,18 +1,130 @@
+# ---------------------------------------------
+# Battery Monitoring Threshold Limits for BMS:
+# ---------------------------------------------    
+# Temperature      : Celsius
+# State of Charge  : Percentage
+# Charge Rate      : Ratio (0 to 1)
+# ---------------------------------------------
+# =============================================================================
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
 
-  return True
+#Declare the battery type currently used
+CURRENT_BATTERYPACK       =   'Lithium'
 
+#Declare operating threshold params for different battery packs
+batteryThresholdParams = { 'Lithium' : 
+                            {
+                            'Temperature'  : { 'lower':  0   , 'upper': 45 },
+                            'StateOfCharge': { 'lower': 20   , 'upper': 80 },
+                            'ChargeRate'   : { 'upper': 0.8                },
+                            }, 
+                                    
+                         'NiMh' :   
+                            {
+                            'Temperature'  : { 'lower': -20  , 'upper': 40 },
+                            'StateOfCharge': { 'lower':  20  , 'upper': 80 },
+                            'ChargeRate'   : { 'upper': 0.8                },
+                            },   
+                         }
+
+def getBatteryThresholdLimit(batteryType):
+    return batteryThresholdParams[batteryType]
+
+#Check if Battery is working fine
+def battery_is_ok(**kwargs):
+  
+  batteryLimits             = getBatteryThresholdLimit(CURRENT_BATTERYPACK)
+  isWithinLimits            = True
+  
+  global BATTERY_CONDITION_ALL_OK
+  BATTERY_CONDITION_ALL_OK  = True
+  
+  for criteria, criteriavalue in kwargs.items():
+      lower,upper    = getBoundaryConditions(batteryLimits,criteria, criteriavalue)
+      isWithinLimits = checkRangeLimits (criteria,criteriavalue,lower,upper)
+       
+      if isWithinLimits is False:
+          BATTERY_CONDITION_ALL_OK = False        
+            
+  return BATTERY_CONDITION_ALL_OK
+
+def getBoundaryConditions(batteryLimits,criteria, criteriavalue):
+      upper = lower = None    
+      upper = batteryLimits[criteria]['upper'] if 'upper' in batteryLimits[criteria] else None
+      lower = batteryLimits[criteria]['lower'] if 'lower' in batteryLimits[criteria] else None
+          
+      return lower,upper 
+
+def checkRangeLimits(criteria,criteriavalue,lower,upper):    
+  
+    if checkLowerLimitBreach(criteriavalue,lower)  or checkUpperLimitBreach(criteriavalue,upper):
+        print ('Alert: {} Threshold Value breached. Current value is {}'.format(criteria,criteriavalue))
```

Comment:
If we need to support multiple languages, the print gets complex. Is it possible to isolate that? And include the low/high in the message?